### ISSUE TITLE
Threads are not fetched after restarting debug adapter

### ISSUE THREAD

**Issue by dlech** [microsoft/vscode#88784]
- VSCode Version: 1.41
- OS Version: Ubuntu 18.04

Steps to Reproduce:

1. Create a debug adapter
2. Implement the pause feature
3. Run the debug adapter and see that pause works
4. Restart the debug adapter using the restart button
5. Observer that pause no longer works

Does this issue occur when all extensions are disabled?: Yes/No

---

The commit https://github.com/microsoft/vscode/pull/44249/commits/d527e92cd62a1a64a3170fbe5e87fa1d1bcc8c67 is used to slow down fetching threads from a debug adapter. However, when using the default restart mechanism, `fetchThreadsScheduler` is disposed (via `dispose(this.rawListeners);` and therefore no longer runs after restarting the debug adapter.

**Comment by isidorn** [comment]
Thanks for filling this issue.
Can you be so kind to fork Mock Debug https://github.com/microsoft/vscode-mock-debug and to create changes on top of it such that this issue is reproducable.
Since I can not reproduce it using node - gif attached

![pause](https://user-images.githubusercontent.com/1926584/72546105-75ba3000-388a-11ea-923f-e923b6262261.gif)

**Comment by dlech** [comment]
It looks like I was missing sending an `InitializedEvent` so the same debug session was being reused after restart.

**Comment by dlech** [comment]
I also made this comment in the pull request:

The [specification](https://microsoft.github.io/debug-adapter-protocol/specification) says "A debug adapter is expected to send this event when it is ready to accept configuration requests (but not before the ‘initialize’ request has finished)."

So either the specification needs to say "required" instead of "expected" or there is a bug in VS Code. VS Code doesn't seem to correctly handle the situation where the debug adapter never sends  `InitializedEvent`. Although the fix may be to create a new debug session object instead of the fix that I have proposed.
